_posts/2024-01-30-securing-wildfly-apps-oidc-k8s.adoc

```diff
@@ -0,0 +1,368 @@
+
+. Create a new client as follows:
+* *General Settings*:
+** *Client type* (or *Client Protocol*, depending on your Keycloak version): *OpenID Connect*
+** *Client ID*: *myclient*
+* *Capability config*:
+** *Authentication flow*: *Standard flow*, *Direct access grants*
+* *Login settings*: Leave the fields blank for now.
+
++
+For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#_oidc_clients[Managing OpenID Connect clients].
+
+. Click *Save* to save the client.
+
+== Build a docker image of the application
+
+. Navigate to the application's directory.
+ 
+
+. Use the `wildfly-maven-plugin` plugin to create a docker image.
++
+[source,options="nowrap"]
+----
+$ mvn package wildfly:image -Popenshift
+ ...
+[INFO] Successfully tagged localhost/simple-webapp-oidc:latest
+[INFO] acd8c6c41788a99c5b22d2c6b2e42ce024a89e2eb4fe5b4d721f9b56796e95bc
+[INFO] Successfully built application image simple-webapp-oidc:latest
+
+----
+
+. Verify that you see the image in Docker images.
++
+[source,options="nowrap"]
+----
+$ docker images
+...
+REPOSITORY                       TAG         IMAGE ID      CREATED        SIZE
+localhost/simple-webapp-oidc     latest      acd8c6c41788  3 minutes ago  690 MB
+----
+
+== Push the just created image to a container registry
+
+. Log in to the container registry.
++
+[source,options="nowrap",subs=+quotes]
+----
+$ docker login __CONTAINER_REGISTRY__ 
+----
++
+Substitute __CONTAINER_REGISTRY__ as follows:
+
+* For Quay, use `quay.io`.
+* For Docker Hub, use `docker.io`.
+
+. Create a new tag for your image.
++
+[source,options="nowrap"]
+----
+$ docker tag simple-webapp-oidc __TAGGED_IMAGE__
+----
++
+Substitute __TAGGED_IMAGE__ as follows:
+
+* For Quay, use the form `quay.io/__USERNAME__/simple-webapp-oidc`.
+* For Docker Hub, use `__USERNAME__/simple-webapp-oidc`.
+
+. Verify that you see the tagged image in Docker images.
++
+[source,options="nowrap"]
+----
+$ docker images
+...
+REPOSITORY                           TAG         IMAGE ID      CREATED         SIZE
+localhost/ashiot/simple-webapp-oidc  latest      acd8c6c41788  20 minutes ago  690 MB
+----
+
+. Push the image to your container registry.
++
+[source,options="nowrap"]
+----
+$ docker push __TAGGED_IMAGE__:latest
+----
++
+Substitute __TAGGED_IMAGE__ as follows:
+
+* For Quay, use the form `quay.io/__USERNAME__/simple-webapp-oidc`.
+* For Docker Hub, use `__USERNAME__/simple-webapp-oidc`.
+
+
+== Add Helm Configuration
+
+. Obtain the URL for Keycloak.
++
+[source]
+----
+KEYCLOAK_URL=http://$(minikube ip):$(kubectl get services/keycloak -o go-template='{{(index .spec.ports 0).nodePort}}') &&
+echo "" &&
+echo "Keycloak URL:   $KEYCLOAK_URL" &&
+echo ""
+----
+
+. Switch to the `charts` directory in the `simple-webapp-oidc` example.
++
+[source]
+----
+cd /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-oidc/charts
+----
+
+. Create a file `values.yml`.
```

Comment:
Hi @fjuma, do we want to add this file in https://github.com/wildfly-security-incubator/elytron-examples/tree/main/simple-webapp-oidc ?

Reply:
Hi @theashiot, apologies for the delayed response. I haven't gone through this in detail yet but yes, adding a file with the required configuration for k8s to the existing example would make sense.

Reply:
Thanks, @fjuma, i've created a PR to add the file:  https://github.com/wildfly-security-incubator/elytron-examples/pull/209

I'll updated the steps shortly.

Reply:
I've updates the steps. Ready for review!

Please note that I haven't tested the steps for Quay. I have based them on https://www.wildfly.org/news/2023/06/16/deploy-on-kubernetes-with-helm/ .  I'm having some authentication problems with Quay.

I've tested the steps for Docker Hub.